tests/rbd/test_scale_group_snap_clone.py

```diff
@@ -0,0 +1,241 @@
+            for image, image_config in pool_config.items():
+                retry = 0
+                while retry < 5:
+                    try:
+                        clone = image + "_clone"
+                        io_config = {
+                            "rbd_obj": rbd,
+                            "client": client,
+                            "size": fio["size"],
+                            "do_not_create_image": True,
+                            "config": {
+                                "file_size": fio["size"],
+                                "file_path": [f"/mnt/mnt_{random_string(len=5)}/file"],
+                                "get_time_taken": True,
+                                "image_spec": [f"{pool}/{clone}"],
+                                "operations": {
+                                    "fs": "ext4",
+                                    "io": True,
+                                    "mount": True,
+                                    "map": True,
+                                },
+                                "skip_mkfs": False,
+                                "cmd_timeout": 2400,
+                                "io_type": "write",
+                            },
+                        }
+
+                        krbd_io_handler(**io_config)
+                        break
+                    except Exception as e:
+                        log.info(
+                            f"Map of the clone {clone} failed: {str(e)} : retrying again.."
+                        )
+                        time.sleep(10)
+                        retry = retry + 1
+                if retry == 5:
+                    log.error("Failure in FIO even after 5 retries")
+
+    return 0
+
+
+def run(**kw):
+    """Tests the group creation followed by cloning image from group snapshot at scale.
+
+    Args:
+        **kw: test data
+    """
+    log.info(
+        "Running test CEPH-83594558 - Cloning an image from a Group Snapshot at scale"
+    )
+
+    try:
+        if kw.get("client_node"):
+            client = get_node_by_id(kw.get("ceph_cluster"), kw.get("client_node"))
+        else:
+            client = kw.get("ceph_cluster").get_nodes(role="client")[0]
+
+        rbd_obj = initial_rbd_config(**kw)
+        pool_types = rbd_obj.get("pool_types")
+
+        ret_val = test_scale_group_snap_clone(rbd_obj=rbd_obj, client=client, **kw)
+        if ret_val == 0:
+            log.info("Testing RBD cloning image from group snapshot at scale Passed")
+
+    except Exception as e:
+        log.error(
+            f"Testing RBD cloning image from group snapshot at scale failed with Error: {str(e)}"
+        )
+        ret_val = 1
+
+    finally:
+        cluster_name = kw.get("ceph_cluster", {}).name
+        if "rbd_obj" not in locals():
+            rbd_obj = Rbd(client)
+        obj = {cluster_name: rbd_obj}
+        if pool_types:
+            cleanup(pool_types=pool_types, multi_cluster_obj=obj, **kw)
```

Comment:
please check this cleanup removes `ec data pool` as well?

Reply:
Done